keps/sig-node/3063-dynamic-resource-allocation/kep.yaml

```diff
@@ -0,0 +1,44 @@
+title: dynamic resource allocation
+kep-number: 3063
+authors:
+  - "@pohly"
+owning-sig: sig-node
+participating-sigs:
+  - sig-scheduling
+status: provisional
+creation-date: 2021-05-17
+reviewers:
+  - "@klueska"
+  - "@ahg-g"
```

Comment:
we need an api review as well

Reply:
Can you add me as well?

Also, is it worth involving a sig-storage reviewer? I'm wondering if they regret any API/architecture decisions that they would recommend we avoid here?

Reply:
> we need an api review as well

Isn't that going to be covered during code review?

> Can you add me as well?

Sure, the more the merrier :grin: 

> Also, is it worth involving a sig-storage reviewer? I'm wondering if they regret any API/architecture decisions that they would recommend we avoid here?

I already mentioned it to SIG Storage and I think @jsafrane looked at the KEP. I can ping them again, but wouldn't want to make the KEP approval depend on that, i.e. not add someone to the kep.yaml.

Reply:
> Isn't that going to be covered during code review?

From experience, the more complex the feature, the better it is to include an API reviewer from the beginning.

Reply:
I agree, that would be good. I just don't want the KEP approval to be blocked by such a review.

How about this:
- SIG Node and Scheduling finish the review
- SIG Node approves
- I prepare a PR with the API changes for a formal API review, without actually merging it yet.
- Depending on the outcome of the API review we
  - proceed with the implementation for 1.24,
  - punt it to 1.25 for further discussion, or
  - (worst case), abandon the effort.

Reply:
I don't think it is only about the API, this is a change that involves three controllers with complex interactions, scheduler, kubelet and controller-manager; an architecture review is needed.

Reply:
An architecture review makes sense to me. Any suggestions how we can arrange that?

Reply:
As discussed with Dims on Slack I sent an email: https://groups.google.com/g/kubernetes-sig-architecture/c/drYdXWp8EsI

Reply:
sg, I see Tim and Derek will be looking at it, sounds good.